File: guides/micronaut-microservices-distributed-tracing-xray/bookinventory/java/src/main/java/example/micronaut/Book.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.validation.constraints.NotBlank;

@Serdeable
public record Book(
        @NotBlank String id,
        @NotBlank String isbn,
        @NotBlank String name,
        @Nullable Integer stock
) implements Identified {

    @Override
    @NonNull
    public String getId() {
        return id;
    }
}

File: guides/micronaut-file-download-excel/python/src/example/micronaut/book.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.core.annotation import Introspected


@Introspected
@dataclass
class Book:
    isbn: Annotated[str, NotBlank]
    name: Annotated[str, NotBlank]
